test/test_automl/test_estimators.py

```diff
@@ -465,6 +466,156 @@ def exclude(lst, s):
                 assert all(leaderboard['ensemble_weight'] > 0)
 
 
+@pytest.mark.parametrize('estimator', [AutoSklearnRegressor])
+@pytest.mark.parametrize('resampling_strategy', ['holdout'])
+@pytest.mark.parametrize('X', [
+    np.asarray([[1.0, 1.0, 1.0]] * 50 + [[2.0, 2.0, 2.0]] * 50)
+])
+@pytest.mark.parametrize('y', [
+    np.asarray([1.0] * 25 + [2.0] * 25 + [3.0] * 25 + [4.0] * 25)
+])
+def test_show_models_with_holdout(
+    tmp_dir: str,
+    dask_client: Client,
+    estimator: AutoSklearnEstimator,
+    resampling_strategy: str,
+    X: np.ndarray,
+    y: np.ndarray
+) -> None:
+    """
+    Parameters
+    ----------
+    tmp_dir: str
+        The temporary directory to use for this test
+
+    dask_client: dask.distributed.Client
+         The dask client to use for this test
+
+    estimator: AutoSklearnEstimator
+         The estimator to train
+
+    resampling_strategy: str
+         The resampling strategy to use
+
+    X: np.ndarray
+        The X data to use for this estimator
+
+    y: np.ndarray
+         The targets to use for this estimator
+
+    Expects
+    -------
+    * Expects all the model dictionaries to have ``model_keys``
+    * Expects all models to have an auto-sklearn wrapped model ``regressor``
+    * Expects all models to have a sklearn wrapped model ``sklearn_regressor``
+    * Expects no model to have any ``None`` value
+    """
+
+    automl = estimator(
+        time_left_for_this_task=60,
+        per_run_time_limit=5,
+        tmp_folder=tmp_dir,
+        resampling_strategy=resampling_strategy,
+        dask_client=dask_client
+    )
+    automl.fit(X, y)
+
+    models = automl.show_models().values()
+
+    model_keys = [
+        'model_id', 'rank', 'cost', 'ensemble_weight',
+        'data_preprocessor', 'feature_preprocessor',
+        'regressor', 'sklearn_regressor'
+    ]
+
+    assert all([model_keys == list(model.keys()) for model in models]) is True
+    assert all([model['regressor'] for model in models]) is True
+    assert all([model['sklearn_regressor'] for model in models]) is True
+    assert any([None in model.values() for model in models]) is False
```

Comment:
I personally prefer using `not` instead of `is False` and just excluding `is True` but it's not a problem though. Change if you wish.

```python
assert all(...)
assert not any(...)
```